modules/webgl/src/adapter/converters/device-parameters.ts

```diff
@@ -156,10 +164,14 @@ export function setDeviceParameters(device: Device, parameters: Parameters) {
     const ext = extensions.WEBGL_provoking_vertex;
 
     if (parameters.provokingVertex) {
-      const vertex = map('provokingVertex', parameters.provokingVertex, {
-        first: GL.FIRST_VERTEX_CONVENTION_WEBGL,
-        last: GL.LAST_VERTEX_CONVENTION_WEBGL
-      });
+      const vertex = map<ProvokingVertex, GLProvokingVertex>(
```

Comment:
Can't we deduce the type?

Reply:
By declaring the type, we force TypeScript errors if any enum values have been missed — note the "lint" -> "line" correction earlier in the file. If we infer types from the inputs, nothing is validated.